src/libcore/convert.rs

```diff
@@ -86,14 +84,13 @@
 /// Using `identity` to keep the `Some` variants of an iterator of `Option<T>`:
```

Comment:
I disagree.

First, I think that `.filter_map(identity)` is clearer wrt. intent (and also semantics). I can clearly tell that filtering is happening because it says so in the name. I know that `.filter_map` takes `f: T -> Option<U>` and with `identity` I fix `T = Option<U>` and thus I get back all the `Some`s. The `.flatten()` method doesn't tell me any of this, in particular you cannot tell without knowing that `Option` is being operated on what the semantics are. Thus the reasoning footprint is greater with `.flatten()`.

Second, even tho `.flatten()` is more general than `join :: Monad m => m (m a) -> m a` I think the primary use of the method should be for monadic join and using it for other purposes will break some people's minds (e.g. mine).

Third, I think that we shouldn't hold up stabilization on this issue; we can resolve it in a subsequent PR or issue if you feel strongly.